app/models/course/level.rb

```diff
@@ -12,6 +16,20 @@ def self.after_course_initialize(course)
     course.levels.build(experience_points_threshold: 0)
   end
 
+  # Make use of RANK(), a postgres window function to generate level numbers (currently not
+  # stored in the database). Raw SQL is used as there is no adapter to generate the SQL query.
+  # Since rank starts from 1 and Course::Levels start from 0, 1 is deducted from rank .
+  #
+  # @return [ActiveRecord::Querying]
+  def self.with_level_number
+    select <<-SQL.strip_heredoc
+    *,  (-1 + rank() OVER
```

Comment:
drop the *, only select the particular field you want.

In your default scope, then do `.select('*')`, separately.